pkg/expression/builtin_compare.go

```diff
@@ -1431,6 +1431,13 @@ func RefineComparedConstant(ctx sessionctx.Context, targetFieldType types.FieldT
 		targetFieldType = *types.NewFieldType(mysql.TypeLonglong)
 	}
 	var intDatum types.Datum
+
+	// To make sure return zero when underflow happens.
+	oriFlag := sc.IsRefineComparedConstant
+	sc.IsRefineComparedConstant = true
```

Comment:
I'm not sure if there is some multip-threads problems if you set this to true?

Reply:
`StatementContext` is per query. Also this is during plan building. I think there is no concurrency issue.

Reply:
OK